tsdb/ooo_head_read.go

```diff
@@ -264,34 +251,34 @@ func (cr OOOHeadChunkReader) ChunkOrIterable(meta chunks.Meta) (chunkenc.Chunk,
 	}
 
 	s.Lock()
-	if s.ooo == nil {
-		// There is no OOO data for this series.
-		s.Unlock()
-		return nil, nil, storage.ErrNotFound
-	}
-	mc, err := s.oooMergedChunks(meta, cr.head.chunkDiskMapper, cr.mint, cr.maxt)
+	mc, err := s.oooMergedChunks(meta, cr.head.chunkDiskMapper, cr.cr, cr.mint, cr.maxt)
 	s.Unlock()
-	if err != nil {
-		return nil, nil, err
-	}
 
-	// This means that the query range did not overlap with the requested chunk.
-	if len(mc.chunkIterables) == 0 {
-		return nil, nil, storage.ErrNotFound
-	}
+	return nil, mc, err
+}
 
-	return nil, mc, nil
+// Pass through special behaviour for current head chunk.
```

Comment:
This comment is unclear, IMHO, and it should start with `ChunkOrIterableWithCopy` to comply with the codestyle.